syntax = "proto3";

package docs;

// The documents saving service definition.
service DocsSaver {
  // Save one document
  rpc SaveDocument (SaveDocumentRequest) returns (SaveDocumentReply) {}
  // Save multiple documents
  rpc SaveDocuments (stream SaveDocumentRequest) returns (SaveDocumentReply) {}
}


message Point {
  float lat = 1;
  float lon = 2;
}

// The request message containing document data
message SaveDocumentRequest {
  int32 article_id = 1;
  string text = 2;
  string title = 3;
  string date = 4;
  string lang = 5;
  repeated Point locations = 6;
  repeated float semantic_vector = 7;
  repeated string keywords = 8;
  repeated string entities = 9;
  repeated string themes = 10;
  repeated string class = 11;
}

// The resposense message containing operation status
message SaveDocumentReply {
  string message = 1;
}
